github/prci.py

```diff
@@ -252,17 +255,22 @@ def check_reboot(self, *args, **kwargs):
             self.reboot = True
 
     def run(self):
-        def join(wait=False):
-            for p in copy.copy(self.processes):
-                if wait or not p.is_alive():
+        def join():
+            for p in self.processes:
+                if not p.is_alive():
                     p.join()
+                    logging.info('Calling join() for process')
                     self.processes.remove(p)
 
         def execute_task(task):
             try:
                 task.execute()
-            except Exception:
+            except Exception as e:
+                logging.error(e)
```

Comment:
We may or may not want `exc_info=True` here. I'm not sure.

Reply:
Actually, I think that here we could have: 
`logging.exception('Task %s failed to execute', task.name, exc_info=True)`